fenicsprecice/fenicsprecice.py

```diff
@@ -162,7 +165,10 @@ def update_coupling_expression(self, coupling_expression, data):
         if not self._empty_rank:
             vertices = np.array(list(data.keys()))
             nodal_data = np.array(list(data.values()))
-            coupling_expression.update_boundary_data(nodal_data, vertices[:, 0], vertices[:, 1])
+            if self._fenics_dims == 2:
+                coupling_expression.update_boundary_data(nodal_data, vertices[:, 0], vertices[:, 1])
+            elif self._fenics_dims == 3:
+                coupling_expression.update_boundary_data(nodal_data, vertices[:, 0], vertices[:, 1], vertices[:, 2])
```

Comment:
Is `update_boundary_data` exposed to the user? If yes, we have to keep this siganture. But we could still provide a generic alternative `update_boundary_data(nodal_data, vertices)`, where `vertices` is a `d x n` array.

Reply:
This is addressed in https://github.com/precice/fenics-adapter/pull/145